firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -251,16 +260,91 @@ void recalculateAndSaveOverlays(Set<DocumentKey> documentKeys) {
     return results;
   }
 
-  /** Queries the remote documents and overlays by doing a full collection scan. */
+  /**
+   * Given a collection group, returns the next documents that follow the provided offset, along
+   * with an updated offset.
+   *
+   * <p>This method returns documents in order of read time from the provided offset, followed by
```

Comment:
Nit: The returned documents are sorted by key. They are only consumed by read time.